src/modules/evaluateExpression/evaluateExpressionTestData.ts

```diff
@@ -0,0 +1,708 @@
+  type: 'number',
+  operator: 'pgSQL',
+  children: [
+    {
+      value: 'SELECT count(*) FROM template',
+    },
+  ],
+};
+
+testData.getListOfTemplates_noType = {
+  operator: 'pgSQL',
+  children: [
+    {
+      value: 'SELECT template_name FROM template',
+    },
+  ],
+};
+
+// GraphQL operator -- TO DO
+
+// More complex combinations
+
+testData.concatFirstAndLastNames = {
+  type: 'string',
+  operator: 'CONCAT',
+  children: [
+    {
+      operator: 'objectProperties',
+      children: [{ value: { object: 'user', property: 'firstName' } }],
+    },
+    {
+      value: ' ',
+    },
+    {
+      operator: 'objectProperties',
+      children: [{ value: { object: 'user', property: 'lastName' } }],
+    },
+  ],
+};
+
+testData.complex1 = {
+  operator: 'AND',
+  children: [
+    {
+      operator: '=',
+      children: [
+        {
+          operator: 'objectProperties',
+          children: [
+            {
+              value: { object: 'form', property: 'q1' },
+            },
+          ],
+        },
+        {
+          value: 'Drug Registration',
+        },
+      ],
+    },
+    {
+      operator: '!=',
+      children: [
+        {
+          type: 'number',
+          operator: 'pgSQL',
+          children: [
+            { value: 'SELECT COUNT(*) FROM user_organisation WHERE user_id = $1' },
+            {
+              operator: 'objectProperties',
+              children: [{ value: { object: 'user', property: 'id' } }],
+            },
+            {
+              operator: 'objectProperties',
+              children: [{ value: { object: 'organisation', property: 'id' } }],
+            },
+          ],
+        },
+        {
+          type: 'number',
+          value: 0,
+        },
+      ],
+    },
+  ],
+};
+
+testData.complex2 = {
+  operator: 'AND',
+  children: [
+    {
+      operator: '=',
+      children: [
+        {
+          operator: 'objectProperties',
+          children: [{ value: { object: 'application', property: 'stage' } }],
+        },
+        {
+          value: 1,
+        },
+      ],
+    },
+    {
+      operator: '=',
+      children: [
+        {
+          operator: 'graphQL',
+          children: [
+            {
+              value:
+                'Graph QL to return COUNT of questions for current application that are NOT Approved',
+            },
+            { value: { object: 'application', property: 'id' } },
+          ],
+        },
+        {
+          value: 0,
+        },
+      ],
+    },
+  ],
+};
+
+testData.complex2_asString = `{
+    "operator": "AND",
+    "children": [
+      {
+        "operator": "=",
+        "children": [
+          {
+            "operator": "objectProperties",
+            "children": [
+              {
+                "value": {
+                  "object": "application",
+                  "property": "stage"
+                }
+              }
+            ]
+          },
+          {
+            "value": 1
+          }
+        ]
```

Comment:
The same problem described in the previous `complex2` query

Reply:
Same answer 😜